youtube_dl/YoutubeDL.py

```diff
@@ -758,6 +758,15 @@ def extract_info(self, url, download=True, ie_key=None, extra_info={},
                                     'and will probably not work.')
 
             try:
+                if self.original_ie:
+                    pass # self.original_ie already exists
+            except AttributeError:
+                try:
+                    self.original_ie = re.search("\.([^\.]+)'>", str(type(ie))).group(1)
```

Comment:
This is a bad idea since it breaks reusing of `YoutubeDL` instances. Even worse it stores string instead of just reference.